xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -2109,16 +2108,7 @@ server4_icreate_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
                  "ICREATE [%s]",
                  frame->root->unique, uuid_utoa(stbuf->ia_gfid));
 
-    link_inode = inode_link(inode, state->loc.parent, state->loc.name, stbuf);
-
-    if (!link_inode) {
-        op_ret = -1;
-        op_errno = ENOENT;
-        goto out;
-    }
-
-    inode_lookup(link_inode);
-    inode_unref(link_inode);
+    inode_link_lookup(inode, state->loc.parent, state->loc.name, stbuf);
```

Comment:
Based on old code, this could potentially fail. Shouldn't we return errors from `inode_link_lookup()` ?

Reply:
Done